docs/design/2020-08-08-recursive-common-table-expression.md

```diff
@@ -0,0 +1,108 @@
+
+# Proposal: Recursive Common Table Expression
+
+- Author(s):     [Pingyu](https://github.com/pingyu) (Ping Yu)
+- Last updated:  2020-08-18
+- Discussion at: https://github.com/pingcap/tidb/issues/17472
+
+
+## Abstract
+
+This proposal proposes to support the __Recursive Common Table Expression (Recursive CTE)__.
+
+
+## Background
+
+From MySQL User Document, [13.2.15 WITH (Common Table Expressions)](https://dev.mysql.com/doc/refman/8.0/en/with.html):
```

Comment:
It's better to describe more about the usage of recursive CTE like what we did in the proposal of non-recursive CTE.